boot/freeldr/freeldr/arch/realmode/helpers.inc

```diff
@@ -153,36 +153,47 @@ Relocator16Boot:
     popf
 
     /* Setup the segment registers */
-    mov ax, word ptr cs:[BSS_RegisterSet + REGS_DS]
+    mov ax, word ptr ss:[BSS_RegisterSet + REGS_DS]
     mov ds, ax
-    mov ax, word ptr cs:[BSS_RegisterSet + REGS_ES]
+    mov ax, word ptr ss:[BSS_RegisterSet + REGS_ES]
     mov es, ax
-    mov ax, word ptr cs:[BSS_RegisterSet + REGS_FS]
+    mov ax, word ptr ss:[BSS_RegisterSet + REGS_FS]
     mov fs, ax
-    mov ax, word ptr cs:[BSS_RegisterSet + REGS_GS]
+    mov ax, word ptr ss:[BSS_RegisterSet + REGS_GS]
     mov gs, ax
 
     /* Patch the jump address (segment:offset) */
-    mov eax, dword ptr cs:[BSS_RealModeEntry]
+    mov eax, dword ptr ss:[BSS_RealModeEntry]
     mov dword ptr cs:[Relocator16Address], eax
 
-    /* Switch the stack (segment:offset) */
+    /* Store the stack (segment:offset) */
     mov eax, dword ptr cs:[BSS_CallbackReturn]
     shr eax, 16
-    mov ss, ax
+    mov word ptr cs:[Relocator16SS], ax
     mov eax, dword ptr cs:[BSS_CallbackReturn]
     and eax, HEX(0FFFF)
-    mov esp, eax
+    mov dword ptr cs:[Relocator16ESP], eax
 
     /* Setup the registers */
-    mov eax, dword ptr cs:[BSS_RegisterSet + REGS_EAX]
-    mov ebx, dword ptr cs:[BSS_RegisterSet + REGS_EBX]
-    mov ecx, dword ptr cs:[BSS_RegisterSet + REGS_ECX]
-    mov edx, dword ptr cs:[BSS_RegisterSet + REGS_EDX]
-    mov esi, dword ptr cs:[BSS_RegisterSet + REGS_ESI]
-    mov edi, dword ptr cs:[BSS_RegisterSet + REGS_EDI]
+    mov eax, dword ptr ss:[BSS_RegisterSet + REGS_EAX]
+    mov ebx, dword ptr ss:[BSS_RegisterSet + REGS_EBX]
+    mov ecx, dword ptr ss:[BSS_RegisterSet + REGS_ECX]
+    mov edx, dword ptr ss:[BSS_RegisterSet + REGS_EDX]
+    mov esi, dword ptr ss:[BSS_RegisterSet + REGS_ESI]
+    mov edi, dword ptr ss:[BSS_RegisterSet + REGS_EDI]
```

Comment:
If you used any other segment register than `ss`, you wouldn't need the extra code below.